### DISCUSSION TITLE
Feature request: Allow setting min/max-width/height with all the options as width

### DISCUSSION THREAD

**Discussion by multiwebinc** [tailwindlabs/tailwindcss#11006] [Ideas]
According to the docs, there are only 5 values for [min-width](https://tailwindcss.com/docs/min-width), and a few more for [max-width](https://tailwindcss.com/docs/max-width), however it is not possible to do, for example `min-width-24`, however you can with `width-24`. I suggest adding all of the same options to the `min`/`max`-`width`/`height` options as well.

--- 

For example, if I want the pink column to be 2/3 of the width:

```html
<div class="flex w-64 bg-slate-300">
  <div>
    Some content that spans more than one line here
  </div>
  <div class="bg-red-400 w-2/3">
    Some really really really really really really really really really really really really really long content
  </div>
</div>
```

But instead, due to how flex works, it is only about 30% of the width. 

![image](https://user-images.githubusercontent.com/901732/231966012-fb067ce6-107e-4e2a-b9f0-f1e83e8da623.png)

If I set the `min-width` to `66.666667%`, it displays as desired, but this is not an option with Tailwind:

![image](https://user-images.githubusercontent.com/901732/231966381-430f64ce-94ad-49af-872f-d7d2f19c0b5f.png)

**Comment by olets** [comment]
Currently you can do this with a small configuration change, extending the min/max-h/w configs with the width config.

In this demo the min/max-h/w classes are safelisted, so you can see the result of the `extend` config in the "Generated CSS" pane's "Utilities" tab: https://play.tailwindcss.com/9xlYBgAkT0?file=config

**Comment by multiwebinc** [comment]
@olets Thanks for the reply. I wasn't aware that you could do that. But I'm just wondering, what is the reasoning behind not having something like that by default? Does it somehow save on file size sent to the client or otherwise decrease performance?

**Reply by olets** [reply]
> what is the reasoning

I don't know the reason. See also https://github.com/tailwindlabs/tailwindcss/discussions/1917#discussioncomment-56244 and https://github.com/tailwindlabs/tailwindcss/discussions/1917#discussioncomment-56251

> Does it somehow save on file size sent to the client or otherwise decrease performance?

No, TW only generates CSS for the classes used. The number of values in the min/max-width/height configurations won't impact file size or performance.
